models/src/test/scala/coop/rchain/models/SortedParHashSetSpec.scala

```diff
@@ -82,4 +79,20 @@ class SortedParHashSetSpec extends FlatSpec with Matchers {
     shs.sortedPars should contain theSameElementsAs (expected)
   }
 
+  it should "be equal when it is equal" in {
+    val elements: Seq[Par] = Seq(
+      Par(
+        ids = Seq(
+          GPrivate(
+            ByteString.copyFrom(Array[Byte](
+              0
+            ))
+          ),
+          GPrivate()
+        ),
+        connectiveUsed = true
+      )
+    )
+    assertEqual(SortedParHashSet(elements), SortedParHashSet(elements))
```

Comment:
Would this test still be passing without the changes you made?

Reply:
Yes, it would still be passing because `SortedParHashSet` was overriding `equals` while `SortedParSet` didn't. The problem is that if someone would remove the overridden `equals` we would get the same hard-to-find bug as with `SortedParSet`.